homeassistant/components/vorwerk/vacuum.py

```diff
@@ -0,0 +1,350 @@
+        _LOGGER.debug("self._state=%s", self._state)
+        if "alert" in self._state:
+            robot_alert = ALERTS.get(self._state["alert"])
+        else:
+            robot_alert = None
+        if self._state["state"] == ROBOT_STATE_IDLE:
+            if self._state["details"]["isCharging"]:
+                self._clean_state = STATE_DOCKED
+                self._status_state = "Charging"
+            elif (
+                self._state["details"]["isDocked"]
+                and not self._state["details"]["isCharging"]
+            ):
+                self._clean_state = STATE_DOCKED
+                self._status_state = "Docked"
+            else:
+                self._clean_state = STATE_IDLE
+                self._status_state = "Stopped"
+
+            if robot_alert is not None:
+                self._status_state = robot_alert
+        elif self._state["state"] == ROBOT_STATE_BUSY:
+            if robot_alert is None:
+                self._clean_state = STATE_CLEANING
+                self._status_state = (
+                    f"{MODE.get(self._state['cleaning']['mode'])} "
+                    f"{ACTION.get(self._state['action'])}"
+                )
+                if (
+                    "boundary" in self._state["cleaning"]
+                    and "name" in self._state["cleaning"]["boundary"]
+                ):
+                    self._status_state += (
+                        f" {self._state['cleaning']['boundary']['name']}"
+                    )
+            else:
+                self._status_state = robot_alert
+        elif self._state["state"] == ROBOT_STATE_PAUSE:
+            self._clean_state = STATE_PAUSED
+            self._status_state = "Paused"
+        elif self._state["state"] == ROBOT_STATE_ERROR:
+            self._clean_state = STATE_ERROR
+            self._status_state = ERRORS.get(self._state["error"])
+
+        self._battery_level = self._state["details"]["charge"]
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def supported_features(self):
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_VORWERK
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the vacuum cleaner."""
+        return self._battery_level
+
+    @property
+    def available(self):
+        """Return if the robot is available."""
+        return self._available
+
+    @property
+    def icon(self):
+        """Return specific icon."""
+        return "mdi:robot-vacuum-variant"
+
+    @property
+    def state(self):
+        """Return the status of the vacuum cleaner."""
+        return self._clean_state
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._robot_serial
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the vacuum cleaner."""
+        data = {}
```

Comment:
Might be better to store everything in something like self._attr and then run the below to update a field if not None, or even as part of the update also updating the self._attr and then just returning self._attr.

Reply:
I removed these ATTR, as they are currently not set in the update method.
I also upgraded the the DataUpgradeCoordinator.